src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -211,9 +211,74 @@ public bool TryGetValue(object key, out object? result)
             DateTime utcNow = UtcNow;
 
             CoherentState coherentState = _coherentState; // Clear() can update the reference in the meantime
-            if (coherentState.TryGetValue(key, out CacheEntry? tmp))
+            if (!coherentState.TryGetValue(key, out CacheEntry? entry))
+            {
+                entry = null;
```

Comment:
Is entry not already null when TryGetValue returns false?

Reply:
4f601494c54bc080224f8b2bb540a004ae0da751